tests/bounded_bls12381.rs

```diff
@@ -0,0 +1,127 @@
+    let dpk_bytes0 = serde_wasm_bindgen::from_value::<BlsKeyPair>(key_pair_js0.clone())
+        .unwrap()
+        .publicKey
+        .unwrap();
+    let dpk0 = DeterministicPublicKey::from(array_ref![dpk_bytes0, 0, G2_COMPRESSED_SIZE]);
+
+    let request = BoundedBlsBignatureRequestContextRequest {
+        issuerPublicKey: dpk_bytes0,
+        proverSecretKey: b"WALLET_MASTER_SECRET".to_vec(),
+        messageCount: 10,
+        nonce: b"dummy nonce".to_vec(),
+    };
+    let js_value = serde_wasm_bindgen::to_value(&request).unwrap();
+    let result = bounded_bls_signature_request(js_value).await;
+    assert!(result.is_ok());
+    let result: BoundedBlsSignatureRequestContextResponse = result.unwrap().try_into().unwrap();
+
+    let request = BoundedBlsSignatureVerifyContextRequest {
+        commitment: result.commitment.clone(),
+        proofOfHiddenMessages: result.proofOfHiddenMessages.clone(),
+        challengeHash: result.challengeHash.clone(),
+        messageCount: 10,
+        publicKey: dpk0,
+        nonce: b"dummy nonce".to_vec(),
+    };
+    let js_value = serde_wasm_bindgen::to_value(&request).unwrap();
+    let res = verify_bounded_bls_signature_request(js_value).await;
+    // unsafe {
+    //     console::log_1(&format!("error: {:#?}", res).into());
+    // }
+    assert!(res.is_ok());
+    let res = res.unwrap();
+    assert!(res.is_truthy());
+
+    let request = BoundedBlsSignatureVerifyContextRequest {
+        commitment: result.commitment.clone(),
+        proofOfHiddenMessages: result.proofOfHiddenMessages.clone(),
+        challengeHash: result.challengeHash.clone(),
+        messageCount: 10,
+        publicKey: dpk0,
+        nonce: b"bad nonce".to_vec(),
+    };
+    let js_value = serde_wasm_bindgen::to_value(&request).unwrap();
+    let res = verify_bounded_bls_signature_request(js_value).await;
+    assert!(res.is_ok());
+    let res = res.unwrap();
+    assert!(res.is_falsy());
+}
+
+#[allow(non_snake_case)]
+#[wasm_bindgen_test]
+pub async fn bounded_bls_sign_tests() {
+    let key_pair_js0 = bls_generate_g2_key(Some(vec![0u8; 16])).await.unwrap();
+    let dpk_bytes0 = serde_wasm_bindgen::from_value::<BlsKeyPair>(key_pair_js0.clone())
+        .unwrap()
+        .publicKey
+        .unwrap();
+
+    let request = BoundedBlsBignatureRequestContextRequest {
+        issuerPublicKey: dpk_bytes0.clone(),
+        proverSecretKey: b"WALLET_MASTER_SECRET".to_vec(),
+        messageCount: 3,
+        nonce: b"dummy nonce".to_vec(),
+    };
+    let js_value = serde_wasm_bindgen::to_value(&request).unwrap();
+    let result = bounded_bls_signature_request(js_value).await;
+    assert!(result.is_ok());
+    let result: BoundedBlsSignatureRequestContextResponse = result.unwrap().try_into().unwrap();
+    let blindingFactor = result.blindingFactor;
+
+    // blind sign
+    let messages = vec![
+        b"Message1".to_vec(),
+        b"Message2".to_vec(),
+        b"Message3".to_vec(),
```

Comment:
```suggestion
        string_to_typed_bytes("Message1"),
        string_to_typed_bytes("Message2"),
        string_to_typed_bytes("Message3"),
```
We should use `string_to_typed_bytes` to express the given messages (i.e., "Message1", "Message2", and "Message3") are string-typed values by prefixing it with `U8_STRING = 0u8`.